agent/acs/handler/task_manifest_handler.go

```diff
@@ -0,0 +1,280 @@
+				seelog.Warnf("Unable to handle taskManifest message [%s]: %v", message.String(), err)
+			}
+		}
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) sendTaskStopVerificationMessage() {
+	for {
+		select {
+		case message := <-taskManifestHandler.messageBufferTaskStopVerificationMessage:
+			if err := taskManifestHandler.sendTaskTerminationVerificationMessage(message); err != nil {
+				seelog.Warnf("Unable to handle taskStopVerification message [%s]: %v", message.String(), err)
+			}
+		case <-taskManifestHandler.ctx.Done():
+			return
+		}
+	}
+}
+
+// compares the list of tasks received in the task manifest message and tasks running on the the instance
+// It returns all the task that are running on the instance but not present in task manifest message task list
+func compareTasks(receivedTaskList []*ecsacs.TaskIdentifier, runningTaskList []*apitask.Task) []*ecsacs.TaskIdentifier {
+
+	tasksToBeKilled := make([]*ecsacs.TaskIdentifier, 0)
+	for _, runningTask := range runningTaskList {
+		// For every task running on the instance check if the task is present in receivedTaskList with the DesiredState
+		// of running, if not add them to the list of task that needs to be stopped
+		if runningTask.GetDesiredStatus() == apitaskstatus.TaskRunning {
+			taskPresent := false
+			for _, receivedTask := range receivedTaskList {
+				if *receivedTask.TaskArn == runningTask.Arn && *receivedTask.DesiredStatus == "RUNNING" {
+					// Task present, does not need to be stopped
+					taskPresent = true
+					break
+				}
+			}
+			if !taskPresent {
+				tasksToBeKilled = append(tasksToBeKilled, &ecsacs.TaskIdentifier{
+					DesiredStatus: aws.String("STOPPED"),
+					TaskArn:       aws.String(runningTask.Arn),
+				})
+			}
+		}
+	}
+	return tasksToBeKilled
+}
+
+func (taskManifestHandler *taskManifestHandler) handleSingleMessageVerificationAck(
+	message *ecsacs.TaskStopVerificationAck) error {
+	seelog.Debugf("Message received from sending back task verification ack: ", message)
+
+	// Ensure that we have received a corresponding task manifest message before
+	if *message.MessageId == taskManifestHandler.messageId {
+		// Reset the message id
+		taskManifestHandler.messageId = ""
+		for _, taskToKill := range message.StopTasks {
+			if *taskToKill.DesiredStatus == "STOPPED" {
+				task, isPresent := taskManifestHandler.taskEngine.GetTaskByArn(*taskToKill.TaskArn)
+				if isPresent {
+					seelog.Info("Killing task: %s", task.Arn)
+					task.SetDesiredStatus(apitaskstatus.TaskStopped)
+					taskManifestHandler.taskEngine.AddTask(task)
+				} else {
+					seelog.Debugf("Task not found on the instance: %s", *taskToKill.TaskArn)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (taskManifestHandler *taskManifestHandler) handleTaskManifestSingleMessage(
+	message *ecsacs.TaskManifestMessage) error {
+
+	taskListManifestHandler := message.Tasks
+	seqNumberFromMessage := *message.Timeline
+	agentSequenceNumber := *taskManifestHandler.seqNumTaskManifest
```

Comment:
this sounds like sequence number of the agent.. can you rename this to something like last or latest sequence number?